File: 1054-ComplementOfBase10Integer/1054-ComplementOfBase10Integer.cpp
// Last updated: 3/30/2026, 2:12:33 PM
class Solution {
public:
    int bitwiseComplement(int n) {
        if(n==0) return 1;
        int result = 0, place = 0;
        while(n>0){
            int rem = n%2;
            if(rem==0){
                result+=pow(2,place);
            }
            place++;
            n/=2;
        }
        return result;
    }
};

File: 1054-ComplementOfBase10Integer/1054-ComplementOfBase10Integer.java
// Last updated: 3/30/2026, 2:12:38 PM
class Solution {
    public int bitwiseComplement(int n) {
        String bin = Integer.toBinaryString(n);
        int result = 0, place = bin.length() - 1;
        for(char ch : bin.toCharArray()){
            if(ch=='1'){
                place--;
                continue;
            }
            else {
                result += (int)Math.pow(2,place);
                //System.out.println(result+ " " + place);
            }
            place--;
        }
        return result;
    }
}
/*
101
1 -> continue -> place = 2 - 1 = 1
0 -> result = 2 ^ 1 = 2
1 -> continue
return 1 nice
*/